[Tutorial4] 非同期処理の実装

k-shimizu3 opened the pull request:
### 課題
 - Tutorial 4
 - 非同期処理（async・await）を学ぶ
 - shared_preferencesを使ったデータの読み書き

---

### 主な実装内容
 - lib/main.dart
課題ページへの遷移ボタンを追加

 - lib/async/async_screen.dart
shared_preferencesを使ったデータの読み書きとデータの表示
編集ダイアログの表示

 - pubspec.yaml
パッケージ (shared_preferences) の追加

---
### 完成イメージ
https://user-images.githubusercontent.com/114896398/196624642-74ff9405-fc2a-4f59-a397-302347555aee.mov

---

### 課題（悩んだこと）
特にありません。

---

### 備考
いつもレビューありがとうございます。
レビューよろしくお願いします！

k-shimizu3 committed:
shared_preferences: ^2.0.12 の追加

k-shimizu3 committed:
Tutorial4の実装

k-shimizu3 committed:
コードの整理

onishitaisei commented on lib/async/async_screen.dart:
```diff
@@ -0,0 +1,125 @@
+  String _age = '未設定';
+  String _birthday = '未設定';
+
+  Future<void> _getSharedPreferences() async {
```
SharedPreferencesのインスタンスを取ってくるだけであればこの命名でも良いと思いますが、やってることがローカルに保存されている名前や年齢といったデータを取得することなのでそれにあった命名をした方が良いかもしれないですね！

k-shimizu3 replied:
ありがとうございます！
確かに、SharedPreferencesのインスタンスのみを取ってきそうな命名でした😅
やっていることに適した命名に修正してみます！

k-shimizu3 replied:
_getUserProperty()に変更する形で、コミットhttps://github.com/kenji-axia/flutter_tutorial/pull/9/commits/64db62e4d4ce6f97544aa781fc0a81fdb8643046 で修正しました！

onishitaisei commented on lib/async/async_screen.dart:
```diff
@@ -0,0 +1,125 @@
+    });
+  }
+
+  Future<void> _setSharedPreferences() async {
```
上記と同様です！

k-shimizu3 replied:
ありがとうございます！
こちらも、_storeUserProperty()へ、コミットhttps://github.com/kenji-axia/flutter_tutorial/pull/9/commits/64db62e4d4ce6f97544aa781fc0a81fdb8643046 で修正を行いました！

onishitaisei commented:
2件コメントしました！

また、実際にテキストを入力する際はキーボードが表示されるのが通常の動きかと思いますので、キーボードを表示した上でダイアログ等の挙動を確認してみると良いですね！

k-shimizu3 committed:
_AsyncScreenStateのメソッド名の変更

_getSharedPreferences()  を _getUserProperty() へ変更
_setSharedPreferences() を _storeUserProperty() へ変更

kenji-axia commented:
@onishitaisei 
勉強になります！
修正とキーボードの挙動を確認いたしましたので、お手隙の際にご確認お願いします！

onishitaisei commented:
LGTM!!
